File: ios/sited/sited/src/SdApi.swift
//
//  SdApi.swift
//  sited
//
//  Created by 谢月甲 on 16/2/4.
//  Copyright © 2016年 noear. All rights reserved.
//

import Foundation

public class SdApi
{
    static var _factory : SdNodeFactory?;
    static var _listener : SdLogListener?;
    
    public static func tryInit(factory:SdNodeFactory, listener:SdLogListener){
        if(_factory == nil){
            _factory = factory;
        }
        
        if(_listener == nil){
            _listener = listener;
        }
    }
    
    public static func check()
    {
        if(_factory == nil || _listener==nil){
            //throw exception();
        }
    }
    
}

File: android/sited/src/main/java/org/noear/sited/SdApi.java
package org.noear.sited;

/**
 * Created by yuety on 15/12/19.
 */
public class SdApi {

    protected static SdNodeFactory _factory;
    protected static SdLogListener _listener;

    public static void tryInit(SdNodeFactory factory, SdLogListener listener) {
        _factory = factory;
        _listener = listener;
    }

    protected static void check() throws Exception {
        if (_factory == null || _listener == null) {
            throw new Exception("未初始化");
        }
    }
}
